### PR TITLE
E_CLASSROOM-277 [FE/BE] Implement Subcategory Functionality in Category Form

### PR DESCRIPTION
**Pull request by Teraaay**
### Issue Link

https://framgiaph.backlog.com/view/E_CLASSROOM-277

### Definition of Done
- [x] User can update the location of the category
- [x] Add/edit category should still be functioning

### Related PR

BE Link: https://github.com/framgia/sph-classroom-els-be/pull/56

### Commands to Run
N/A
#### Main note
- Since the category being displayed are the root categories only, to check whether a subcategory is successfully added, you can try to access it in the location modal if it was added in the right parent category. The same when you try to edit a parent category and make it a subcategory.
#### Pages Affected
- ADD A CATEGORY: http://localhost:3003/admin/add-category
- EDIT A CATEGORY: http://localhost:3003/admin/edit-category/1

### Screenshots
ADDING A CATEGORY:
> ADD A CATEGORY

EDITING A CATEGORY:
> EDIT A CATEGORY

### CODE REVIEW

**Comment by ErickDoroy** [comment]
## APPROVE #117
- [x] Able to update location of the category
-----------
- Total Test Time: 0.25 hr
- Total Iterations: 0
- Total Comments: 0

File: src/pages/admin/categories/AddEditCategory/index.js [hunk 1]
```diff
@@ -24,19 +24,47 @@ const AddEditCategory = () => {
...
+  useEffect(() => {
+    if (isSaved) {
+      setParentCategoryID(location?.id);
+      setLocation(null);
+    }
+  }, [location, isSaved]);
+
   useEffect(() => {
     if (loc.pathname !== '/admin/add-category') {
       CategoryApi.show({ categoryId: category_id }).then(({ data }) => {
```
Comment (karlolee50) [inline src/pages/admin/categories/AddEditCategory/index.js]: please put a catch statement

File: src/pages/admin/categories/AddEditCategory/index.js [hunk 2]
```diff
@@ -24,19 +24,47 @@ const AddEditCategory = () => {
...
         setCategory(data.data);
+        setParentCategoryID(data.data.category_id);
+        if (data.data.category_id) {
+          CategoryApi.getParentCategories(category_id).then(({ data }) => {
```
Comment (karlolee50) [inline src/pages/admin/categories/AddEditCategory/index.js]: please put a catch statement

**Review by karlolee50 — CHANGES_REQUESTED** [review]
Review time: 15 minutes
Comment/s: 2

**Pull request merged**